crates/saya/core/src/lib.rs

```diff
@@ -165,42 +164,150 @@ impl Saya {
+
             previous_block_state_root = last_state_root;
 
             // Updating the local state sequentially, as there is only one instance of
             // `self.blockchain` This part does no actual  proving, so should not be a
             // problem
-            for p in params {
-                self.process_block(&mut prove_scheduler, block, p)?;
 
-                if prove_scheduler.is_full() {
-                    self.process_proven(prove_scheduler).await?;
-
-                    prove_scheduler = Scheduler::new(
-                        self.config.batch_size,
-                        self.config.world_address,
-                        prover_identifier.clone(),
-                    );
+            match self.config.mode {
+                SayaMode::Persistent => {
+                    let num_blocks = params.len() as u64;
+                    let calls = params
+                        .into_iter()
+                        .enumerate()
+                        .map(|(i, p)| self.process_block(block + i as u64, p))
+                        .collect::<Result<Vec<_>, _>>()?
+                        .into_iter()
+                        .flatten()
+                        .flat_map(|(_, c)| c)
+                        .map(|c| BatcherCall {
+                            to: c.to,
+                            selector: c.selector,
+                            calldata: c.calldata,
+                            starknet_messages: Vec::new(), // TODO: Fill messages.
+                            appchain_messages: Vec::new(),
+                        })
+                        .collect::<Vec<_>>();
+
+                    let _input = BatcherInput {
+                        calls,
+                        block_number: Felt::from(block),
+                        prev_state_root: mock_state_hash,
+                        block_hash: Felt::from(0u64),
+                    };
+
+                    mock_state_hash += Felt::ONE;
+
+                    info!(target: LOG_TARGET, "Proving {} blocks.", num_blocks);
+
+                    // We might want to prove the signatures as well.
+                    // let proof = self.prover_identifier.prove_snos(input).await?;
+
+                    let input = StarknetOsOutput {
+                        initial_root: mock_state_hash,
+                        final_root: mock_state_hash + Felt::ONE,
+                        prev_block_number: Felt::from(block),
+                        new_block_number: Felt::from(block) + Felt::ONE,
+                        prev_block_hash: Felt::from(1u64),
+                        new_block_hash: Felt::from(2u64),
+                        os_program_hash: Felt::from(42u64),
+                        starknet_os_config_hash: Felt::from(4u64),
+                        use_kzg_da: Felt::from(5u64),
+                        full_output: Felt::from(6u64),
+                        messages_to_l1: vec![],
+                        messages_to_l2: vec![],
```

Comment:
For a next PR, but messages must be extracted from the fetched block's information. They are actually extracted in the `process_block` function.